### PR TITLE
Implement the ktx keyvalue feature for read/write

### PR DESCRIPTION
**Pull request by samcake**
And use it to store meta data needed to reconstruct a gpu::Texture as saved.

### CODE REVIEW

**Comment by samcake** [comment]
@zzmp This should be enough that isf it s part of the branch you can recover the missing bits for the texture.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1b036b7225665ee7e052d33c7b84396bea2c87b0 available at: https://gist.github.com/772d7a243557913cf7966ca4714705b7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9821-1b036b7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9821-1b036b7.dmg

**Comment by jherico** [comment]
This should have an update to the ktx test project to verify loading and saving of the values.

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -1617,8 +1617,8 @@ void MyAvatar::postUpdate(float deltaTime) {
         }
     }
 
-    DebugDraw::getInstance().updateMyAvatarPos(getPosition());
-    DebugDraw::getInstance().updateMyAvatarRot(getOrientation());
+//    DebugDraw::getInstance().updateMyAvatarPos(getPosition());
```
Comment (jherico) [inline interface/src/avatar/MyAvatar.cpp]: Is this meant to go in this PR?  Why are we preserving the lines if they're no longer needed.

**Comment by samcake** [comment]
Just fixed the commented lines, please review, thanks

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4b4f573f8ea682208a0f66b381b4a83cc65f29e8 available at: https://gist.github.com/772d7a243557913cf7966ca4714705b7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9821-4b4f573.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9821-4b4f573.dmg

**Pull request merged**